<a href="https://colab.research.google.com/github/Bgail-ranu/Data-science/blob/main/Warehouse_Data_Handling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Upload File**

In [1]:
from google.colab import files
uploaded = files.upload()

Saving Warehouse_and_Retail_Sales.csv to Warehouse_and_Retail_Sales.csv


# **Part 1**

## **Read**

In [2]:
import pandas as pd
df = pd.read_csv('Warehouse_and_Retail_Sales.csv')

,YEAR,MONTH,SUPPLIER,ITEM CODE,ITEM DESCRIPTION,ITEM TYPE,RETAIL SALES,RETAIL TRANSFERS,WAREHOUSE SALES
0,2020,1,REPUBLIC NATIONAL DISTRIBUTING CO,100009,BOOTLEG RED - 750ML,WINE,0.00,0.0,2.0
1,2020,1,PWSWN INC,100024,MOMENT DE PLAISIR - 750ML,WINE,0.00,1.0,4.0
2,2020,1,RELIABLE CHURCHILL LLLP,1001,S SMITH ORGANIC PEAR CIDER - 18.7OZ,BEER,0.00,0.0,1.0
3,2020,1,LANTERNA DISTRIBUTORS INC,100145,SCHLINK HAUS KABINETT - 750ML,WINE,0.00,0.0,1.0
4,2020,1,DIONYSOS IMPORTS INC,100293,SANTORINI GAVALA WHITE - 750ML,WINE,0.82,0.0,0.0


## **Inspect**

In [14]:
df.head()

,YEAR,MONTH,SUPPLIER,ITEM CODE,ITEM DESCRIPTION,ITEM TYPE,RETAIL SALES,RETAIL TRANSFERS,WAREHOUSE SALES
0,2020,1,REPUBLIC NATIONAL DISTRIBUTING CO,100009,BOOTLEG RED - 750ML,WINE,0.00,0.0,2.0
1,2020,1,PWSWN INC,100024,MOMENT DE PLAISIR - 750ML,WINE,0.00,1.0,4.0
2,2020,1,RELIABLE CHURCHILL LLLP,1001,S SMITH ORGANIC PEAR CIDER - 18.7OZ,BEER,0.00,0.0,1.0
3,2020,1,LANTERNA DISTRIBUTORS INC,100145,SCHLINK HAUS KABINETT - 750ML,WINE,0.00,0.0,1.0
4,2020,1,DIONYSOS IMPORTS INC,100293,SANTORINI GAVALA WHITE - 750ML,WINE,0.82,0.0,0.0


In [4]:
df.tail()

,YEAR,MONTH,SUPPLIER,ITEM CODE,ITEM DESCRIPTION,ITEM TYPE,RETAIL SALES,RETAIL TRANSFERS,WAREHOUSE SALES
307640,2020,9,LEGENDS LTD,99753,DUTCHESS DE BOURGOGNE NR - 750ML,BEER,0.00,0.0,5.00
307641,2020,9,ANHEUSER BUSCH INC,9997,HOEGAARDEN 4/6NR - 12OZ,BEER,66.12,37.0,240.75
307642,2020,9,COASTAL BREWING COMPANY LLC,99970,DOMINION OAK BARREL STOUT 4/6 NR - 12OZ,BEER,2.25,0.0,0.00
307643,2020,9,BOSTON BEER CORPORATION,99990,SAM ADAMS SUMMER VARIETY 12PK NR,BEER,20.50,0.0,0.00
307644,2020,9,NaN,WC,WINE CREDIT,REF,0.00,0.0,-70.00


In [7]:
df.shape

(307645, 9)

In [8]:
df.columns

Index(['YEAR', 'MONTH', 'SUPPLIER', 'ITEM CODE', 'ITEM DESCRIPTION',
       'ITEM TYPE', 'RETAIL SALES', 'RETAIL TRANSFERS', 'WAREHOUSE SALES'],
      dtype='object')

In [12]:
df.isnull().sum()

,0
YEAR,0
MONTH,0
SUPPLIER,167
ITEM CODE,0
ITEM DESCRIPTION,0
ITEM TYPE,1
RETAIL SALES,3
RETAIL TRANSFERS,0
WAREHOUSE SALES,0


In [11]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
YEAR,307645.0,2018.438525,1.083061,2017.00,2017.0,2019.00,2019.0000,2020.00
MONTH,307645.0,6.423862,3.461812,1.00,3.0,7.00,9.0000,12.00
RETAIL SALES,307642.0,7.024071,30.986238,-6.49,0.0,0.32,3.2675,2739.00
RETAIL TRANSFERS,307645.0,6.936465,30.237195,-38.49,0.0,0.00,3.0000,1990.83
WAREHOUSE SALES,307645.0,25.294597,249.916798,-7800.00,0.0,1.00,5.0000,18317.00


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307645 entries, 0 to 307644
Data columns (total 9 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   YEAR              307645 non-null  int64  
 1   MONTH             307645 non-null  int64  
 2   SUPPLIER          307478 non-null  object 
 3   ITEM CODE         307645 non-null  object 
 4   ITEM DESCRIPTION  307645 non-null  object 
 5   ITEM TYPE         307644 non-null  object 
 6   RETAIL SALES      307642 non-null  float64
 7   RETAIL TRANSFERS  307645 non-null  float64
 8   WAREHOUSE SALES   307645 non-null  float64
dtypes: float64(3), int64(2), object(4)
memory usage: 21.1+ MB


In [15]:
df['YEAR']

,YEAR
0,2020
1,2020
2,2020
3,2020
4,2020
...,...
307640,2020
307641,2020
307642,2020
307643,2020


In [20]:
df.loc[:,['WAREHOUSE SALES', 'ITEM TYPE']]

,WAREHOUSE SALES,ITEM TYPE
0,2.00,WINE
1,4.00,WINE
2,1.00,BEER
3,1.00,WINE
4,0.00,WINE
...,...,...
307640,5.00,BEER
307641,240.75,BEER
307642,0.00,BEER
307643,0.00,BEER


In [22]:
df.groupby('ITEM TYPE')['WAREHOUSE SALES'].agg(['sum', 'mean', 'max', 'count'])

,sum,mean,max,count
ITEM TYPE,,,,
BEER,6527236.51,153.897072,18317.0,42413
DUNNAGE,-121454.00,-1278.463158,-1.0,95
KEGS,118431.00,11.672679,604.0,10146
LIQUOR,94906.27,1.462121,438.0,64910
NON-ALCOHOL,26149.59,13.705236,297.0,1908
REF,-20499.00,-161.409449,298.0,127
STR_SUPPLIES,0.00,0.000000,0.0,405
WINE,1156984.91,6.165982,662.0,187640


The full size of the dataset is: 307645 by 9

# **Part 2**

## **A.**

In [38]:
df.loc[:,['WAREHOUSE SALES', 'RETAIL SALES']]

,WAREHOUSE SALES,RETAIL SALES
0,2.00,0.00
1,4.00,0.00
2,1.00,0.00
3,1.00,0.00
4,0.00,0.82
...,...,...
307640,5.00,0.00
307641,240.75,66.12
307642,0.00,2.25
307643,0.00,20.50


In [25]:
df.loc[:,['SUPPLIER', 'ITEM TYPE']]

,SUPPLIER,ITEM TYPE
0,REPUBLIC NATIONAL DISTRIBUTING CO,WINE
1,PWSWN INC,WINE
2,RELIABLE CHURCHILL LLLP,BEER
3,LANTERNA DISTRIBUTORS INC,WINE
4,DIONYSOS IMPORTS INC,WINE
...,...,...
307640,LEGENDS LTD,BEER
307641,ANHEUSER BUSCH INC,BEER
307642,COASTAL BREWING COMPANY LLC,BEER
307643,BOSTON BEER CORPORATION,BEER


## **B.**

In [31]:
filtered_df = df[df['WAREHOUSE SALES'] > 10000]
filtered_df

,YEAR,MONTH,SUPPLIER,ITEM CODE,ITEM DESCRIPTION,ITEM TYPE,RETAIL SALES,RETAIL TRANSFERS,WAREHOUSE SALES
1363,2020,1,CROWN IMPORTS,23445,CORONA EXTRA LOOSE NR - 12OZ,BEER,679.00,766.00,14408.50
13281,2020,7,CROWN IMPORTS,23445,CORONA EXTRA LOOSE NR - 12OZ,BEER,1174.00,1041.00,18317.00
13757,2020,7,HEINEKEN USA,23886,HEINEKEN LOOSE NR - 12OZ,BEER,628.00,586.00,10167.88
23083,2020,7,CROWN IMPORTS,96750,CORONA EXTRA 2/12 NR - 12OZ,BEER,705.00,516.00,10639.50
24572,2020,3,CROWN IMPORTS,23445,CORONA EXTRA LOOSE NR - 12OZ,BEER,992.00,1127.00,10605.00
36694,2017,6,CROWN IMPORTS,23445,CORONA EXTRA LOOSE NR - 12OZ,BEER,1249.00,1160.00,13723.71
48175,2017,6,CROWN IMPORTS,96750,CORONA EXTRA 2/12 NR - 12OZ,BEER,53.00,61.00,15955.46
61204,2017,7,CROWN IMPORTS,96750,CORONA EXTRA 2/12 NR - 12OZ,BEER,75.00,76.00,14411.50
63383,2017,8,CROWN IMPORTS,23445,CORONA EXTRA LOOSE NR - 12OZ,BEER,990.00,1077.00,15346.71
74791,2017,8,CROWN IMPORTS,96750,CORONA EXTRA 2/12 NR - 12OZ,BEER,59.50,56.00,11417.92


In [35]:
filtered1_df = df[(df['WAREHOUSE SALES'] > 5000) & (df['RETAIL SALES'] < 50)]
filtered1_df

,YEAR,MONTH,SUPPLIER,ITEM CODE,ITEM DESCRIPTION,ITEM TYPE,RETAIL SALES,RETAIL TRANSFERS,WAREHOUSE SALES
48188,2017,6,HEINEKEN USA,96970,HEINEKEN 2/12 NR - 12OZ,BEER,39.50,38.0,7288.42
61218,2017,7,HEINEKEN USA,96970,HEINEKEN 2/12 NR - 12OZ,BEER,48.50,40.0,6038.00
63172,2017,8,CROWN IMPORTS,23314,CORONA EXTRA 18PK NR - 12OZ,BEER,0.00,0.0,5220.00
74804,2017,8,HEINEKEN USA,96970,HEINEKEN 2/12 NR - 12OZ,BEER,40.00,46.0,8641.48
88216,2017,9,HEINEKEN USA,96970,HEINEKEN 2/12 NR - 12OZ,BEER,35.00,44.0,5325.00
101833,2017,10,HEINEKEN USA,96970,HEINEKEN 2/12 NR - 12OZ,BEER,48.54,44.0,7205.00
190778,2019,3,MILLER BREWING COMPANY,50714,MILWAUKEE BEST ICE 2/15PK CAN 12OZ,BEER,47.00,48.5,5137.37
208990,2019,5,CROWN IMPORTS,23314,CORONA EXTRA 18PK NR - 12OZ,BEER,0.00,0.0,5245.00
220341,2019,5,CROWN IMPORTS,96741,CORONA EXTRA 4/6 NR - 12OZ,BEER,1.56,0.0,6159.46
232671,2019,6,CROWN IMPORTS,96741,CORONA EXTRA 4/6 NR - 12OZ,BEER,0.84,0.0,5270.67


In [43]:
df.loc[
    (df['WAREHOUSE SALES'] > 5000) & (df['RETAIL SALES'] < 100),
    ['ITEM TYPE', 'SUPPLIER']
    ]

,ITEM TYPE,SUPPLIER
48175,BEER,CROWN IMPORTS
48188,BEER,HEINEKEN USA
61204,BEER,CROWN IMPORTS
61218,BEER,HEINEKEN USA
63172,BEER,CROWN IMPORTS
74791,BEER,CROWN IMPORTS
74804,BEER,HEINEKEN USA
88203,BEER,CROWN IMPORTS
88216,BEER,HEINEKEN USA
101820,BEER,CROWN IMPORTS


In [44]:
df.iloc[:5, :3]

,YEAR,MONTH,SUPPLIER
0,2020,1,REPUBLIC NATIONAL DISTRIBUTING CO
1,2020,1,PWSWN INC
2,2020,1,RELIABLE CHURCHILL LLLP
3,2020,1,LANTERNA DISTRIBUTORS INC
4,2020,1,DIONYSOS IMPORTS INC


In [45]:
df.iloc[10:21, 2:5]

,SUPPLIER,ITEM CODE
10,HEAVEN HILL DISTILLERIES INC,10120
11,BACCHUS IMPORTERS LTD,10123
12,BACCHUS IMPORTERS LTD,10124
13,BACCHUS IMPORTERS LTD,10125
14,MONSIEUR TOUTON SELECTION,101346
15,"THE COUNTRY VINTNER, LLC DBA WINEBOW",101486
16,"THE COUNTRY VINTNER, LLC DBA WINEBOW",101532
17,ROYAL WINE CORP,101664
18,REPUBLIC NATIONAL DISTRIBUTING CO,101702
19,ROYAL WINE CORP,101753


# **Part 3**

## **A.**

In [47]:
df.groupby('ITEM TYPE')['WAREHOUSE SALES'].sum()

,WAREHOUSE SALES
ITEM TYPE,
BEER,6527236.51
DUNNAGE,-121454.00
KEGS,118431.00
LIQUOR,94906.27
NON-ALCOHOL,26149.59
REF,-20499.00
STR_SUPPLIES,0.00
WINE,1156984.91


## **B.**

In [48]:
df.groupby('ITEM TYPE')['WAREHOUSE SALES'].agg(['sum', 'mean', 'max', 'count'])

,sum,mean,max,count
ITEM TYPE,,,,
BEER,6527236.51,153.897072,18317.0,42413
DUNNAGE,-121454.00,-1278.463158,-1.0,95
KEGS,118431.00,11.672679,604.0,10146
LIQUOR,94906.27,1.462121,438.0,64910
NON-ALCOHOL,26149.59,13.705236,297.0,1908
REF,-20499.00,-161.409449,298.0,127
STR_SUPPLIES,0.00,0.000000,0.0,405
WINE,1156984.91,6.165982,662.0,187640


## **C.**

From the result, Beer and wine account for most of the sales, with the highest totals and transaction counts, making them the main drivers in the dataset. Kegs and liquor contribute much less in comparison. Some items like dunnage and REF have negative sales, likely due to returns or adjustments. Overall, sales are concentrated in a few major product types.
